lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -114,35 +148,33 @@ def get(self):
             # Ignore anything that doesn't use our API prefix, because it's
             # not in our API.
             if url.startswith(self.uri_prefix):
-                # If the URI is parameterized with a Flask "<type:name>"
-                # template string, we don't want to report it, so we remove
-                # it from the URI. We derive an API name by converting the
-                # "/" characters in the URI to "_", after removing a trailing
-                # "/" that would have been left by removing a template... we
-                # don't remove the trailing "/" from the URI, which serves as
-                # an indication that the parameter is needed.
-                #
-                # E.g, "/api/v1/controllers/list" yields:
-                #     "controllers_list": "/api/v1/controllers/list"
-                #
-                # while "/api/v1/users/<string:username>" yields:
-                #     "users": "/api/v1/users/"
-                #
-                # TODO: This won't work right with embedded template strings,
-                # which we're not currently using anywhere; but it'll require
-                # adjustment later if we add any. (E.g., something like
-                # "/api/v1/foo/<string:name>/detail/<string:param>")
+                simplified = self.param_template.sub(r"/{\g<name>}", url)
+                template: Dict[str, Any] = {"template": urljoin(host, simplified)}
+                params = []
+                for match in self.param_template.finditer(url):
+                    params.append(
+                        {"name": match.group("name"), "type": match.group("type")}
+                    )
+                template["params"] = params
```

Comment:
Alternately:
```
                params = []
                for match in self.param_template.finditer(url):
                    params.append(
                        {"name": match.group("name"), "type": match.group("type")}
                    )
                template: Dict[str, Any] = {"template": urljoin(host, simplified), "params": params}
```
or, even,
```
                template: Dict[str, Any] = {
                    "template": urljoin(host, simplified),
                    "params": [
                        {"name": match.group("name"), "type": match.group("type")}
                        for match in self.param_template.finditer(url):
                    ]
                }
```
Although, I would pull for something more like this:
```
                template: Dict[str, Any] = {
                    "template": urljoin(host, simplified),
                    "params": {
                        match.group("name"): {"type": match.group("type")}
                        for match in self.param_template.finditer(url):
                    }
                }
```
🙂